lib/screens/language_domain/language_game_three.dart

```diff
@@ -0,0 +1,267 @@
+  final Uri dictionaryUrl =
+      Uri.parse('https://vietnamese-wordlist.duyet.net/Viet74K.txt');
+
+  final int answerDurationInSeconds = 60;
+  final int pointPerCorrectAnswer = 200;
+
+  Duration answerDuration = const Duration();
+  Timer? countdownTimer;
+
+  TextEditingController controller = TextEditingController();
+  late Future<String> firstCharacter;
+  List<String> _answer = [];
+  int _point = 0;
+  GameStatus _status = GameStatus.playing;
+
+  // Timer Handler
+  void startTimer() {
+    answerDuration = Duration(seconds: answerDurationInSeconds);
+
+    countdownTimer =
+        Timer.periodic(const Duration(seconds: 1), (_) => setCountDown());
+  }
+
+  void setCountDown() {
+    const reduceSecondsBy = 1;
+    setState(() {
+      final seconds = answerDuration.inSeconds - reduceSecondsBy;
+      if (seconds < 0) {
+        setCancelTimer();
+        changeStatus(GameStatus.end);
+      } else {
+        answerDuration = Duration(seconds: seconds);
+      }
+    });
+  }
+
+  void setCancelTimer() {
+    countdownTimer!.cancel();
+  }
+
+  // Fetch Data
+  Future<String> fetchRandomCharacter() async {
+    final response = await http.get(dictionaryUrl);
+    final data = response.body;
+    List<String> dataList = data.toLowerCase().split('\n');
+
+    if (dataList.isNotEmpty) {
+      String firstCharacter =
+          dataList[Random().nextInt(dataList.length)].split(' ')[0];
+      _answer.add(firstCharacter);
+      return firstCharacter;
+    }
+
+    throw Exception('Failed to load random character in the dictionary');
+  }
+
+  Future<bool> checkValidWord(String word) async {
+    Map<String, String> headers = {'Content-type': 'application/json'};
+    final response = await http.post(Uri.parse('$baseUrl/check'),
+        headers: headers, body: jsonEncode({'text': word}));
+    if (response.statusCode == 200) {
+      return true;
+    }
+    return false;
+  }
+
+  // Logic Handler
+  void handleClickCheck() async {
+    String userAnswer = controller.text;
+    String firstChar = _answer.last;
+    String checkingWord = '$firstChar $userAnswer';
+
+    bool isValidWord = await checkValidWord(checkingWord);
+    if (isValidWord) {
+      _showToast('Chính xác', Colors.green);
+      _answer.add(userAnswer);
+      setState(() {
+        _point += pointPerCorrectAnswer;
+        // Restart timer
+        answerDuration = Duration(seconds: answerDurationInSeconds);
+      });
+    } else {
+      _showToast('Không hợp lệ', Colors.red);
+    }
+
+    controller.text = '';
+  }
+
+  void changeStatus(GameStatus status) {
+    handleStatusChange(status);
+
+    setState(() {
+      _status = status;
+    });
+  }
+
+  void handleStatusChange(GameStatus status) {
+    if (status == GameStatus.end) {
+      _showMyDialog('Kết thúc', () {
+        Navigator.of(context).pop();
+      });
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    firstCharacter = fetchRandomCharacter();
```

Comment:
should not do this because assign a variable when the app mount every widget which some thing the same as the idea of 
`useEffect( ( ) => {
// someThing....
})`

We need to dipose this assign too